ament_cmake_export_dependencies/cmake/ament_cmake_export_dependencies-extras.cmake.in

```diff
@@ -36,15 +36,21 @@ if(NOT _exported_dependencies STREQUAL "")
 
           get_target_property(_imported_configurations ${_target} IMPORTED_CONFIGURATIONS)
           if(_imported_configurations)
-            get_target_property(_imported_implib ${_target} IMPORTED_IMPLIB_${_imported_configurations})
-            if(_imported_implib)
-              list(APPEND _libraries "${_imported_implib}")
-            else()
-              get_target_property(_imported_location ${_target} IMPORTED_LOCATION_${_imported_configurations})
-              if(_imported_location)
-                list(APPEND _libraries "${_imported_location}")
+            foreach(_imported_config ${_imported_configurations})
+              set(_imported_implib_config "optimized")
+              get_target_property(_imported_implib ${_target} IMPORTED_IMPLIB_${_imported_config})
+              if(_imported_implib STREQUAL "DEBUG")
```

Comment:
Should this condition take [DEBUG_CONFIGURATIONS](https://cmake.org/cmake/help/v3.19/prop_gbl/DEBUG_CONFIGURATIONS.html) into account (if available)?

Reply:
Hi @dirk-thomas ,

Yes it should,  changed it to take into account the  DEBUG_CONFIGURATIONS global variable as suggested, and also had a bug, fixed it too, maybe I need to squash the commits.

Thanks,
Csaba

Reply:
> maybe I need to squash the commits.

No need to. We do squash on merge anyway.